templates/named.conf.options.erb

```diff
@@ -1,5 +1,5 @@
 options {
-	directory "/var/cache/bind";
+	directory "<%= scope.lookupvar('::dns::server::params::data_dir') %>";
```

Comment:
For the first point:  The PR includes the change to the dns::server::options class:
  ) {
    $valid_check_names = ['fail', 'warn', 'ignore']
    $cfg_dir = $::dns::server::params::cfg_dir
+  $data_dir = $::dns::server::params::data_dir

    if ! defined(Class['::dns::server']) {
      fail('You must include the ::dns::server base class before using any dns options defined resources')

For your second point:  Overriding this is not currently possible using hiera because it's hardcoded in the template,  unless you're saying that to make this work on RedHat, every user must define their data_dir to be /var/cache/named.

- Johnson

Paulo Edgar Castro wrote:
> In templates/named.conf.options.erb <https://github.com/ajjahn/puppet-dns/pull/118#discussion_r31315448>:
>
>> @@ -1,5 +1,5 @@
>>  options {
>> -	directory "/var/cache/bind";
>> +	directory "<%= scope.lookupvar('::dns::server::params::data_dir') %>";
>
> Changing this do data_dir will involve either changing the dns::server::options to accept another param or creating a local variable within the module that is assigned the ::dns::server::params::data_dir variable.
>
> Overriding this is already possible by using hiera and passing in params into the params module.
>
> Your call ...
>
> —
> Reply to this email directly or view it on GitHub <https://github.com/ajjahn/puppet-dns/pull/118/files#r31315448>.
>

-- 
Johnson Earls
darkfoxprime@gmail.com